### PR TITLE
clipping: clip a segment

### PR DESCRIPTION
**Pull request by emranemran**
Part 2 in series of Clipping related PRs. 

This change adds logic to take any segment and clip it to the start/end time specified and also re-encodes it with some x264 args to re-encode and place I-frames at the right intervals. 

TODO: an integration test will be added in a separate PR.

### CODE REVIEW

File: video/clip.go [hunk 1]
```diff
@@ -86,3 +98,24 @@ func ClipManifest(requestID string, manifest *m3u8.MediaPlaylist, startTime, end
...
+	err := ffmpeg.Input(tsInputFile).
+		Output(tsOutputFile, ffmpeg.KwArgs{"ss": start, "to": end,
+			"c:v":          "libx264",
+			"r":            "24",
+			"g":            "48",
+			"keyint_min":   "48",
+			"sc_threshold": 50,
+			"bf":           "0",
+			"c:a":          "copy"}).
+		OverWriteOutput().ErrorToStdOut().Run()
```
Comment (emranemran) [inline video/clip.go:115]: Note: these settings will likely be revisited to be optimized before we release clipping. Using these reasonable defaults for now to finish integration.

File: video/clip.go [hunk 2]
```diff
@@ -4,8 +4,20 @@ import (
 	"fmt"
 	"github.com/grafov/m3u8"
 	"github.com/livepeer/catalyst-api/log"
+	ffmpeg "github.com/u2takey/ffmpeg-go"
+	"time"
 )
 
+// format time in secs to be copatible with ffmpeg's expected time syntax
+func formatTime(seconds float64) string {
```
Comment (thomshutt) [inline video/clip.go:12]: This works, but Go does have pretty good builtin time formatting

File: video/clip.go [hunk 3]
```diff
@@ -86,3 +98,24 @@ func ClipManifest(requestID string, manifest *m3u8.MediaPlaylist, startTime, end
...
 }
+
+// Clips a segment by re-encoding so that I-frames are placed at the right intervals
+func ClipSegment(tsInputFile, tsOutputFile string, startTime, endTime float64) error {
+	start := formatTime(startTime)
+	end := formatTime(endTime)
+	err := ffmpeg.Input(tsInputFile).
+		Output(tsOutputFile, ffmpeg.KwArgs{"ss": start, "to": end,
+			"c:v":          "libx264",
+			"r":            "24",
```
Comment (thomshutt) [inline video/clip.go:109]: Could we have some comments explaining all these values please?

Comment (emranemran) [reply video/clip.go:99]: Done

File: video/clip.go (hunk 2 above)
Comment (emranemran) [reply video/clip.go:12]: Updated to use the builtin methods.

**Review by thomshutt — APPROVED** [review]

**Pull request merged**